src/pytorch_lightning/loggers/mlflow.py

```diff
@@ -238,18 +238,14 @@ def experiment_id(self) -> Optional[str]:
     def log_hyperparams(self, params: Union[Dict[str, Any], Namespace]) -> None:
         params = _convert_params(params)
         params = _flatten_dict(params)
-        params_list: List[Param] = []
 
-        for k, v in params.items():
-            # TODO: mlflow 1.28 allows up to 500 characters: https://github.com/mlflow/mlflow/releases/tag/v1.28.0
-            if len(str(v)) > 250:
-                rank_zero_warn(
-                    f"Mlflow only allows parameters with up to 250 characters. Discard {k}={v}", category=RuntimeWarning
-                )
-                continue
-            params_list.append(Param(key=k, value=v))
+        # Truncate parameter values to 250 characters.
+        # TODO: MLflow 1.28 allows up to 500 characters: https://github.com/mlflow/mlflow/releases/tag/v1.28.0
+        params_list = [Param(key=k, value=str(v)[:250]) for k, v in params.items()]
```

Comment:
What are your thoughts about the issue of colliding keys, raised by @schmidt-jake 
https://github.com/Lightning-AI/lightning/pull/16418#issuecomment-1397522199

Reply:
@awaelchli keys are not truncated, so that won't be an issue... unless we want to truncate the keys too. I guess it's not as common to have a parameter name that's more than 250 characters. We can also exclude parameters whose name is longer than 250 characters and log a warning message, if we want, but that wasn't done earlier either.